### PR TITLE
Setup skeleton for pkgdown auto-generating R docs

### PR DESCRIPTION
**Pull request by terrytangyuan**
@angela0xdata @ledell The diff of this PR is pretty hard to visualize but I basically helped put together a skeleton for pkgdown files with some common practices, including a tab for tutorials and a tab for API references (only has random forest linked there for now but you can add more in the `pkgdown.yml` file). 

You can view the website by opening `h2o-3/h2o-r/h2o-package/docs/index.html` and regenerate the website by executing `pkgdown::build_site("~/h2o-3/h2o-r/h2o-package/")`. Most of the content is just a placeholder for now though.

### CODE REVIEW

**Comment by ledell** [comment]
Hey @terrytangyuan I think some files got included in here by accident.  There are some model files and zip files: e.g. `h2o-r/h2o-package/docs/reference/GBM_model_R_1512747612023_635.zip`  Can you look through the files and see if there are other binary/zip files that can be removed?

File: h2o-r/h2o-package.template [hunk 1]
```diff
@@ -28,7 +28,7 @@ Note that no actual data is stored in the R workspace; and no actual work is car
 If you are using an older version of H2O, use the following porting guide to update your scripts: \href{https://github.com/h2oai/h2o-dev/blob/master/h2o-docs/src/product/upgrade/H2ODevPortingRScripts.md}{Porting Scripts}
 }
 \author{
-Anqi Fu, Tom Kraljevic and Petr Maj, with contributions from the H2O.ai team
+The H2O.AI team
```
Comment (ledell) [inline h2o-r/h2o-package.template]: Should be H2O.ai not H2O.AI

File: h2o-r/h2o-DESCRIPTION.template [hunk 2]
```diff
@@ -3,7 +3,7 @@ Version: SUBST_PROJECT_VERSION
 Type: Package
 Title: R Interface for H2O
 Date: SUBST_PROJECT_DATE
-Author: The H2O.ai team
+Author: The H2O.AI team
```
Comment (ledell) [inline h2o-r/h2o-DESCRIPTION.template]: Should be H2O.ai not H2O.AI

**Comment by ledell** [comment]
@terrytangyuan This is a great start!  We will need to also make some modifications to the H2O build process to include re-generating these docs and inserting the correct version number into the docs.  

Once we have come up with some intro tutorials (using [keras](https://keras.rstudio.com/index.html) for inspiration) and finished landing pages, then we can work with @angela0xdata to figure out a place to host this on our docs.h2o.ai website.

**Comment by angela0xdata** [comment]
Hmm... I'm getting an error when I try to build:
```
> pkgdown::build_site("~/Desktop/repos/h2o-3/h2o-r/h2o-package/")
Initialising site --------------------------------------------------------------------
Error: Current working directory, '/Users/techwriter',  does not appear to be inside a project or package.
```

**Comment by terrytangyuan** [comment]
@angela0xdata Seems like need to be in that directory first. This should work:
```
setwd("~/h2o-3/h2o-r/h2o-package")
pkgdown::build_site(".")
```

**Comment by terrytangyuan** [comment]
@ledell Good catch! I just addressed your comments and squashed commits.

**Comment by angela0xdata** [comment]
Can we ignore all of the output files in h2o-r/h2o-package/docs?

**Comment by terrytangyuan** [comment]
@angela0xdata Done. Just removed and ignored that folder.

**Comment by terrytangyuan** [comment]
@h2o-ops please test

**Pull request merged**